tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts

```diff
@@ -1,135 +1,14 @@
-        // apiVersionTexts.length must be 0 or 1, otherwise the binding pattern contains the same keys, which causes a ts error
-        if (apiVersionTexts.length === 1 && apiVersionTexts[0]) {
-            apiVersion = extractApiVersionFromText(apiVersionTexts[0]);
-        }
-    });
-    return apiVersion;
-};
-
-// workaround for createClient function changes it's way to setup api-version
-export const findApiVersionInRestClient = (clientPath: string): string | undefined => {
-    const version2 = findApiVersionInRestClientV2(clientPath);
-    if (version2) {
-        return version2;
-    }
-    const version1 = findApiVersionInRestClientV1(clientPath);
-    return version1;
-};
-
-const getApiVersionTypeFromRestClient: IApiVersionTypeExtractor = (
-    packageRoot: string
-): ApiVersionType => {
-    const clientPath = findRestClientPath(packageRoot);
-    const apiVersion = findApiVersionInRestClient(clientPath);
-    if (apiVersion && apiVersion.indexOf("-preview") >= 0)
-        return ApiVersionType.Preview;
-    if (apiVersion && apiVersion.indexOf("-preview") < 0)
-        return ApiVersionType.Stable;
-    return ApiVersionType.None;
-};
-
-const findApiVersionsInOperations = (
-    sourceFile: SourceFile | undefined
-): Array<string> | undefined => {
-    const interfaces = sourceFile?.getInterfaces();
-    const interfacesWithApiVersion = interfaces?.filter((itf) =>
-        itf.getProperty('"api-version"')
-    );
-    const apiVersions = interfacesWithApiVersion?.map((itf) => {
-        const property = itf.getMembers().filter((m) => {
-            const defaultValue = m.getChildrenOfKind(
-                SyntaxKind.StringLiteral
-            )[0];
-            return defaultValue && defaultValue.getText() === '"api-version"';
-        })[0];
-        const apiVersion = property
-            .getChildrenOfKind(SyntaxKind.LiteralType)[0]
-            .getText();
-        return apiVersion;
-    });
-    return apiVersions;
-};
-
-const getApiVersionTypeFromOperations: IApiVersionTypeExtractor = (
-    packageRoot: string
-): ApiVersionType => {
-    const paraPath = path.join(packageRoot, "src/rest/parameters.ts");
-    const sourceFile = getTsSourceFile(paraPath);
-    const apiVersions = findApiVersionsInOperations(sourceFile);
-    if (!apiVersions) return ApiVersionType.None;
-    const previewVersions = apiVersions.filter(
-        (v) => v.indexOf("-preview") >= 0
-    );
-    return previewVersions.length > 0
-        ? ApiVersionType.Preview
-        : ApiVersionType.Stable;
-};
+import { findParametersPath, findRestClientPath, getApiVersionTypeFromOperations, getApiVersionTypeFromRestClient } from "../../xlc/apiVersion/utils";
 
 export const getApiVersionType: IApiVersionTypeExtractor = (
     packageRoot: string
 ): ApiVersionType => {
-    const typeFromClient = getApiVersionTypeFromRestClient(packageRoot);
+    const relativeRestSrcFolder = "src/rest/";
```

Comment:
> I guess we need to change this part if emitter has next release as Timo's PR has been merged.

yes, it can, the ci will build the package before test. deleted

Reply:
the new emitter is released last friday.